### PR TITLE
Task, process and terminal tests clean up

### LINKED ISSUE #1784: `Unhandled promise rejections` in git tests seems to break OSx build
It is not reproducible with `npx run test @theia/git`, but with `yarn test:theia` i get the following error:
```
(node:16921) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
error An unexpected error occurred: "Command failed.
Exit code: 1
Command: sh
Arguments: -c run test \"@theia/!(example-)*\" --parallel
Directory: /Users/kosyakov/oomph/theia-master/git/theia
Output:
".
```

See https://travis-ci.org/theia-ide/theia/jobs/371392558

### PR DESCRIPTION
**Pull request by akosyakov**
This PR:
- fixes #1784 
- make sure that task, process and terminal tests run with isolated containers and clean up after themselves

### CODE REVIEW

File: packages/core/src/node/messaging/test/test-web-socket-channel.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as ws from 'ws';
+import * as http from 'http';
+import * as https from 'https';
+import { WebSocketChannel } from '../../../common/messaging/web-socket-channel';
+import { Disposable } from 'vscode-jsonrpc';
```
Comment (kittaakos) [inline packages/core/src/node/messaging/test/test-web-socket-channel.ts]: It should be `@theia/core/lib/common/disposable` or something like that, right?

File: packages/terminal/src/node/terminal-backend-contribution.slow-spec.ts [hunk 2]
```diff
@@ -1,18 +1,17 @@
 /*
- * Copyright (C) 2017 Ericsson and others.
+ * Copyright (C) 2018 Ericsson and others.
```
Comment (kittaakos) [inline packages/terminal/src/node/terminal-backend-contribution.slow-spec.ts]: Minor, we need to decide whether we do this or not. I have seen this in other PRs too. I would rather leave as is, or write `2017, 2018` instead.

**Comment by akosyakov** [comment]
@kittaakos I've fixed `Disposable` import, is it good to approve?

**Comment by akosyakov** [comment]
@kittaakos https://ci.appveyor.com/project/kittaakos/theia/build/1.0.3330 seems to fail because of other reasons, i never see such errors from nsfw on travis

**Comment by akosyakov** [comment]
An issue for nsfw failures on win: https://github.com/theia-ide/theia/issues/1799

**Pull request merged**